Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/SANS/SANSTubeCalibration.py

```diff
@@ -200,6 +208,12 @@ def validateInputs(self):
         if self.getProperty(Prop.END_PIXEL).value > DetectorInfo.NUM_PIXELS_IN_TUBE:
             issues[Prop.END_PIXEL] = f"The ending pixel must be less than or equal to {DetectorInfo.NUM_PIXELS_IN_TUBE}"
 
+        cvalues_filepath = self.getProperty(Prop.CVALUE_FILE).value
+        if cvalues_filepath:
+            cvalues_filepath = self._set_filepath_extension(cvalues_filepath, self._CVALUES_FILE_EXT)
+            if Path(cvalues_filepath).is_file():
+                issues[Prop.CVALUE_FILE] = "The CValues file already exists"
```

Comment:
It might be good to check here that the chosen location is writable. If you set the path to `/` or  something read-only then the saving only fails at the very end, after the fairly long processing time.

Reply:
I know what you mean, I had the same thought, but I couldn't seem to find a nice way of doing this in Python that didn't involve opening the file - is there a way to do it that I'm missing?

Just for context, the failure at the end will still mean the calibration result is valid and can be used, it just means the cvalues won't have written to file. The instrument scientist could copy and past the values from the messages pane, as they do at the moment, so there is a reasonable workaround if we can't find a nice way to validate this up front.

Reply:
os.access(cvalues_filepath, os.W_OK) would do it. There's a bit of complexity with windows occasionally having file paths that wouldn't get caught by this, so it's not perfect and shouldn't replace the checks that use `with` when we actually create the file. But it would catch most of the most common cases like trying to put the file in `C:\` or `/`

Reply:
Thanks, I hadn't spotted that. I've just had a go with it and it seems to be returning false for filepaths that I'm able to write to on my machine, though. I tried testing outside the algorithm, directly in the Python REPL, and had the same result.

Reply:
Interesting. I wonder if that's a windows issue, as it seems to work okay on my machine. As there's an easy workaround I'm happy to consider this resolved as it's not that major. 

It may be worth adding a mention in the error message that the same values can be found in the log and copied into a file manually as a compromise.

Reply:
I realised on Windows that you have to use a raw string to get it to work (otherwise Windows backslashes in the filepath are an issue). Irrespective of that, though, [this article](https://stackoverflow.com/questions/25908743/why-a-file-is-writtable-but-os-access-file-os-w-ok-return-false) seems to be suggesting that the file has to exist to be able to check whether it's writable, which is what I'm finding on my machine I'm afraid. So I think we might need to do without this.

On adding to the log, I think I'd rather not delay this PR for that, if that's OK with you - the cvalue output in the messages pane is easy to spot and the docs for the algorithm mention that it's printed there. In addition, the algorithm is being used by SANS instrument scientists who become familiar with it's output and have their own usage instructions, so I think just making it clear what has gone wrong and allowing them to decide what to do about it feels sufficient here.

Reply:
Sure, that's fine by me